tests/integration/client/sdk/v1/test_datasets.py

```diff
@@ -65,6 +65,38 @@ async def test_list_datasets(role: UserRole) -> None:
     assert isinstance(response.parsed[0], FeedbackDatasetModel)
 
 
+@pytest.mark.parametrize(
+    "role, with_workspace, expected_length",
+    [
+        (UserRole.owner, False, 2),
+        (UserRole.owner, True, 1),
+        (UserRole.admin, False, 0),
+        (UserRole.admin, True, 1),
+        (UserRole.annotator, False, 0),
+        (UserRole.annotator, True, 1),
+    ],
+)
+@pytest.mark.asyncio
+async def test_list_datasets_by_workspace_id(role: UserRole, with_workspace: bool, expected_length: int) -> None:
+    workspace = await WorkspaceFactory.create()
+    dataset = await DatasetFactory.create(workspace=workspace)
+    user = await UserFactory.create(role=role, workspaces=[dataset.workspace] if with_workspace else [])
+
+    another_workspace = await WorkspaceFactory.create()
```

Comment:
(Just a comment) we should start thinking about a seed script to feed integration test data in a different way

Reply:
With "a different way" do you mean without using the factories? Ideally we can define a function on each test suite (we are already doing that in some places) to call a `self` function for e.g. create a dataset (with different args), to call that function instead of explicit calls to the factories